packages/core/src/integrations/inboundfilters.ts

```diff
@@ -36,198 +35,184 @@ export class InboundFilters implements Integration {
-        // The bug is caused by multiple SDK instances, where one is minified and one is using non-mangled code.
-        // Unfortunatelly we cannot fix it reliably (thus reserved property in rollup's terser config),
-        // as we cannot force people using multiple instances in their apps to sync SDK versions.
-        const options = typeof self._mergeOptions === 'function' ? self._mergeOptions(clientOptions) : {};
-        if (typeof self._shouldDropEvent !== 'function') {
-          return event;
+      if (hub) {
+        const self = hub.getIntegration(InboundFilters);
+        if (self) {
+          const client = hub.getClient();
+          const clientOptions = client ? client.getOptions() : {};
+          const options = _mergeOptions(self._options, clientOptions);
+          return _shouldDropEvent(event, options) ? null : event;
         }
-        return self._shouldDropEvent(event, options) ? null : event;
       }
       return event;
     });
   }
+}
 
-  /** JSDoc */
-  private _shouldDropEvent(event: Event, options: Partial<InboundFiltersOptions>): boolean {
-    if (this._isSentryError(event, options)) {
-      if (isDebugBuild()) {
-        logger.warn(`Event dropped due to being internal Sentry Error.\nEvent: ${getEventDescription(event)}`);
-      }
-      return true;
+/** JSDoc */
+export function _mergeOptions(
+  intOptions: Partial<InboundFiltersOptions> = {},
+  clientOptions: Partial<InboundFiltersOptions> = {},
+): Partial<InboundFiltersOptions> {
+  return {
+    allowUrls: [
+      // eslint-disable-next-line deprecation/deprecation
+      ...(intOptions.whitelistUrls || []),
+      ...(intOptions.allowUrls || []),
+      // eslint-disable-next-line deprecation/deprecation
+      ...(clientOptions.whitelistUrls || []),
+      ...(clientOptions.allowUrls || []),
+    ],
+    denyUrls: [
+      // eslint-disable-next-line deprecation/deprecation
+      ...(intOptions.blacklistUrls || []),
+      ...(intOptions.denyUrls || []),
+      // eslint-disable-next-line deprecation/deprecation
+      ...(clientOptions.blacklistUrls || []),
+      ...(clientOptions.denyUrls || []),
+    ],
+    ignoreErrors: [...(intOptions.ignoreErrors || []), ...(clientOptions.ignoreErrors || []), ...DEFAULT_IGNORE_ERRORS],
+    ignoreInternal: typeof intOptions.ignoreInternal !== 'undefined' ? intOptions.ignoreInternal : true,
+  };
+}
+
+/** JSDoc */
+export function _shouldDropEvent(event: Event, options: Partial<InboundFiltersOptions>): boolean {
+  if (_isSentryError(event, options.ignoreInternal)) {
+    if (isDebugBuild()) {
+      logger.warn(`Event dropped due to being internal Sentry Error.\nEvent: ${getEventDescription(event)}`);
     }
-    if (this._isIgnoredError(event, options)) {
-      if (isDebugBuild()) {
-        logger.warn(
-          `Event dropped due to being matched by \`ignoreErrors\` option.\nEvent: ${getEventDescription(event)}`,
-        );
-      }
-      return true;
+    return true;
+  }
```

Comment:
Rather than passing the option and checking it in `_isSentryError` (and it potentially turning effectively into a no-op), I think it would be better to check it here, and not even call the function if the question is made moot by the option's value.

(That said, do you know why we even have this option? It's not documented, and it's not used anywhere in any sentry repo. Could we just get rid of it?)

Reply:
We use this to drop all `SentryError` errors, which is used pretty extensively throughout the codebase. I don't think we want people to turn this off.

For example, we throw a `SentryError` if we don't parse a DSN properly.

> I think it would be better to check it here, and not even call the function if the question is made moot by the option's value

Great point, will change.

Reply:
> I don't think we want people to turn this off.

Exactly. And we don't anywhere. So I'm asking why we need to have an off switch at all

Reply:
This is a good question. I'm going to open up a follow up issue - but leave it like this for now.